docs/source/conf.py

```diff
@@ -123,13 +123,46 @@
     AttributeDocumenter,
     Documenter,
     MethodDocumenter,
+    PropertyDocumenter,
 )
 from sphinx.ext.autosummary import Autosummary  # isort:skip
 
 
+class ModinAccessorDocumenter(PropertyDocumenter):
+    """
+    Generates documentation for properties of Modin objects like Series.str and Series.dt that
+    are themselves accessor classes.
+    This class is necessary because we need to monkeypatch the Series.str/dt property objects
+    with the actual classes in order for autosummary-generate to produce stubs for them.
+
+    This class is not responsible for properties of those accessors like Series.str.capitalize.
+
+    See sphinx source for PropertyDocumenter:
+    https://github.com/sphinx-doc/sphinx/blob/907d27dc6506c542c11a7dd16b560eb4be7da5fc/sphinx/ext/autodoc/__init__.py#L2691
+    """
+
+    objtype = "modinaccessor"
+    directivetype = "attribute"
+
+    # lower priority than the other custom attribute documenter (defined below)
+    priority = 0.55
+
+    def import_object(self, raiseerror=False):
+        import modin.pandas as pd
```

Comment:
can you add comment here about what are we doing? it is not obvious here, for example why self.object_name = self.objpath[-1]?

Reply:
added a comment